### PR TITLE
Removal of Third Party Auth

### PR DESCRIPTION
**Pull request by jessicamack**
##### SUMMARY

As referenced here: https://forum.ansible.com/t/transitioning-authentication-and-authorization-rbac-to-the-new-awx-architecture

##### ISSUE TYPE

 - Breaking Change 

##### COMPONENT NAME

 - API
 - UI
 - Collection
 - CLI
 - Docs
 - Other

##### AWX VERSION

```

```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

File: awx/api/views/__init__.py [hunk 1]
```diff
@@ -684,30 +684,18 @@ class AuthView(APIView):
...
-                    saml_backend_data['login_url'] = '%s?idp=%s' % (login_url, idp)
-                    full_backend_name = '%s:%s' % (name, idp)
-                    if (err_backend == full_backend_name or err_backend == name) and err_message:
-                        saml_backend_data['error'] = err_message
-                    data[full_backend_name] = saml_backend_data
-            else:
-                if err_backend == name and err_message:
-                    backend_data['error'] = err_message
-                data[name] = backend_data
+        if 'ansible_base.authentication' in getattr(settings, "INSTALLED_APPS", []):
```
Comment (hakbailey) [inline awx/api/views/__init__.py]: I still think we want to remove this entire view, as a view in AWX listing single-sign-on endpoints no longer makes sense.

File: docs/docsite/rst/release_notes/known_issues.rst [hunk 2]
```diff
@@ -14,7 +14,6 @@ Known Issues
    pair: known issues; live event statuses
    pair: live event statuses; green dot
    pair: live event statuses; red dot
-   pair: known issues; LDAP authentication
```
Comment (hakbailey) [inline docs/docsite/rst/release_notes/known_issues.rst:17]: We should also remove these two below:
```
   pair: known issues; login problems with social authentication
   pair: known issues; OAuth account recreation
```

**Review by hakbailey — CHANGES_REQUESTED** [review]
In addition to the inline comments, we should remove the [part of the requirements README that references social-auth-app-django](https://github.com/ansible/awx/blob/feature_3rd-party-auth-removal/requirements/README.md#social-auth-app-django), which has been removed.

File: pre-commit.sh [hunk 3]
```diff
@@ -1,14 +1,14 @@
...
+# if [ -z $AWX_IGNORE_BLACK ] ; then
+# 	python_files_changed=$(git diff --cached --name-only --diff-filter=AM awx/ awxkit/ tools/ | grep -E '\.py$')
+# 	if [ "x$python_files_changed" != "x" ] ; then
+#         	black --check $python_files_changed || \
+# 		if [ $? != 0 ] ; then
+#         		echo 'To fix this, run `make black` to auto-format your code prior to commit, or set AWX_IGNORE_BLACK=1'
+#         		exit 1
+# 		fi
+# 	fi
+# fi
```
Comment (TheRealHaoLiu) [inline pre-commit.sh]: please revert this change

**Review by hakbailey — APPROVED** [review]

**Comment by chrismeyersfsu** [comment]
I see sso app migrations deleted. I can see this working for a new install but does it break upgrading?

I looked for, but didn’t find, the deletion of the sso app from the Django app list. On mobile right now, going to look again on the desktop and update my comment.

edit: I see that the `awx.sso` app has been removed from the app list :+1: and @TheRealHaoLiu helped me find the special sauce migration https://github.com/ansible/awx/pull/15571/files#diff-55acf9c639221d7558ed496c210bfc76ec7ee5a037faae48adcc5a000499fc22 :+1: 

I approve this PR.

**Review by chrismeyersfsu — APPROVED** [review]

**Pull request merged**